homeassistant/components/unifiprotect/sensor.py

```diff
@@ -53,50 +52,54 @@ class ProtectSensorEntityDescription(ProtectRequiredKeysMixin, SensorEntityDescr
 
     precision: int | None = None
 
+    def get_ufp_value(self, obj: ProtectAdoptableDeviceModel | NVR) -> Any:
+        """Return value from UniFi Protect device."""
+        value = super().get_ufp_value(obj)
+
+        if isinstance(value, float) and self.precision:
+            value = round(value, self.precision)
+        return value
+
+
+def _get_uptime(obj: ProtectAdoptableDeviceModel | NVR) -> datetime | None:
+    if obj.up_since is None:
+        return None  # pragma: no cover
+
+    # up_since can vary slightly over time
+    # truncate to ensure no extra state_change events fire
+    return obj.up_since.replace(second=0, microsecond=0)
+
+
+def _get_nvr_recording_capacity(obj: Any) -> int:
+    assert isinstance(obj, NVR)
+
+    if obj.storage_stats.capacity is None:
+        return 0  # pragma: no cover
```

Comment:
Unless this is unreachable we should avoid the no cover

Reply:
I can probably write a test case for this one now that I think about after a nights sleep.